packages/babel-types/src/definitions/utils.ts

```diff
@@ -96,11 +96,18 @@ export function assertEach(callback: Validator): Validator {
   function validator(node: t.Node, key: string, val: any) {
     if (!Array.isArray(val)) return;
 
-    for (let i = 0; i < val.length; i++) {
-      const subkey = `${key}[${i}]`;
+    let i = 0;
+    // For performance
+    const subKey = {
+      toString() {
+        return `${key}[${i}]`;
+      },
+    };
```

Comment:
Can you expand a bit the comment here?
- concatenating the strings is expensive because we are actually concatenating a string and a number, so V8 cannot just create a "rope string" but has to allocate memory for the string resulting from the number
- this string is very rarely used, only in error paths, so we can skip the concatenation cost in most cases